### PR TITLE
[bug-fix] Normalization update with SeparateActorCritic, SAC bugfixes

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

Fix two issues with hybrid actions:

- Normalization of the critic was not being updated with SeparateActorCritic. 
- In SAC, stop gradient on Q function during value update, and other small performance tweaks. 

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/sac/optimizer_torch.py [hunk 1]
```diff
@@ -261,33 +261,39 @@ def sac_value_loss(
...
-                    ),
-                    dim=0,
-                )
-                _q2p_mean = torch.mean(
-                    torch.stack(
-                        [torch.sum(_br, dim=1, keepdim=True) for _br in _branched_q2p]
-                    ),
-                    dim=0,
-                )
+            for name in values.keys():
```
Comment (andrewcoh) [inline ml-agents/mlagents/trainers/sac/optimizer_torch.py:264]: Not sure why this is part of the diff

**Review by andrewcoh — APPROVED** [review]
:shipit:

File: ml-agents/mlagents/trainers/sac/optimizer_torch.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/sac/optimizer_torch.py:264]: It's indented so this is under a stop gradient
